glances_api/__init__.py

```diff
@@ -161,4 +161,14 @@ async def get_ha_sensor_data(self) -> dict[str, Any] | None:
             sensor_data["raid"] = data
         if data := self.data.get("uptime"):
             sensor_data["uptime"] = data
+        if data := self.data.get("gpu"):
+            sensor_data["gpu"] = {}
+            for sensor in data:
+                sensor_data["gpu"][f"GPU_{sensor['gpu_id']}__{sensor['name']}"] = {
+                    "name": sensor["name"],
+                    "temperature": sensor["temperature"],
+                    "mem": sensor["mem"],
+                    "proc": sensor["proc"],
+                    "fan_speed": sensor["fan_speed"] if "fan_speed" in sensor else 0,
+                }
```

Comment:
What do you think of using more descriptive names here in the glances-api package already instead of doing the translation in Home Assistant?
```suggestion
                unique_name = f"GPU_{sensor['gpu_id']}__{sensor['name']}"
                sensor_data["gpu"][unique_name] = {
                    "name": sensor["name"],
                    "memory_use_percent": round(sensor["mem"], 1),
                    "processor_use_percent": sensor["proc"],
                    "temperature_celsius": sensor["temperature"],
                    "fan_speed_percent": sensor.get("fan_speed", 0),
                }
```

Reply:
I assume that adding the unit of measurement to a sensor name could be problematic as I'm frankly not sure if the return value could depend on the system's settings (Celsius vs. Fahrenheit, etc.).

Reply:
The Home Assistant interface will need to display a unit anyways so I figured the close to the source we resolve the ambiguity, the better.

I based myself on the comments in this part of the glances codebase to be sure about the units:
https://github.com/nicolargo/glances/blob/develop/glances/plugins/gpu/model.py#L278

A bit further up in that same file, I see a mention of Fahrenheit conversion but that seems to be only for the GUI.

Reply:
API  HOST 61208/api/3/gpu  
[{"key":"gpu_id","gpu_id":0,"name":"Tesla P4","mem":26.248677571614582,"proc":0,"temperature":42,"fan_speed":null}]